### PR TITLE
Upgrade Conviva integration to be compatible with V3 version of Bitmovin iOS Player SDK

### PR DESCRIPTION
**Pull request by luckygoyal-bitmovin**
### Problem
A new major version V3 of Bitmovin iOS SDK was release in April 2021. All new integrations need to use V3 SDK and all new feature requests also will be available only in V3 player SDK. So need to update the Bitmovin Conviva integration to use V3 SDK to allow existing and new customers to use this with V3 player SDK

### Solution
Upgrading Bitmovin Conviva integration to use V3 version of Bitmovin iOS player SDK

### Notes
- Changes to use V3 layer SDK API
- Updated test cases to use new Player API
- Conviva integration is functionally same as old version. No new feature or fix is added to the same.
- Playlist feature support is not added yet to Conviva integration

### Checklist
- [ ] I added an update to `CHANGELOG.md` file
- [ ] I ran all the tests in the project and they succeed

### CODE REVIEW

**Comment by luckygoyal-bitmovin** [comment]
Please review the PR. Following items are in progress and will update PR with same but sharing the PR already for review as there are a lot of changes and may take sometime to review and address review comments.
- Currently 5 test cases out of total 85 are failing. These are related to `PlayerError` and `SourceError` are failing. These are because with V3 SDK it is not possible to create PlayerErrorEvent and SourceErrorEvent objects in application so the test cases cannot fake these events. Trying to find another way to implemented these tests
- Readme and Change log

**Review by thomhs22 — COMMENTED** [review]
Implementation looks good, as far as naming convention and refactor of new variable names.

**Comment by luckygoyal-bitmovin** [comment]
> Implementation looks good, as far as naming convention and refactor of new variable names.

Thanks @thomhs22 , I have updated CHANGELOG and README now

**Review by thomhs22 — APPROVED** [review]
reviewed update on Readme. Looks good

**Comment by stonko1994** [comment]
> > 
> > 
> > Yes, I saw that but still wanted to ask especially. I would propose to end the session when a `PlaylistTransitionEvent` occurs to not report state from subsequent Sources. And of course, stating this in the Limitations in the Readme.
> 
> Do you suggest this to ensure that if any customer uses Playlist API with this Conviva integration, then we should ensure that session atleast ends after first source in playlist? I would rather keep it unhandled yet.

Yes that would be my suggestion 🤔 Hmm we could also keep it unhandled, however, my thought is rather having the first source in a playlist tracked (and ended/stoped) properly than continuing tracking events from a subsequent source. But I leave it up to you at this point, you might have more insights into how this should be handled in the first version.

File: Example/Tests/Doubles/BitmovinPlayerTestDouble.swift
Comment (stonko1994) [inline Example/Tests/Doubles/BitmovinPlayerTestDouble.swift:113]: This cast to `Ad` should also not be needed.

File: Example/Tests/Doubles/BitmovinPlayerTestDouble.swift
Comment (stonko1994) [inline Example/Tests/Doubles/BitmovinPlayerTestDouble.swift:120]: This cast to `Ad` should also not be needed.

File: README.md
Comment (stonko1994) [inline README.md:96]: Just a suggestion to make clear what this is about.

```suggestion
- Tracking multiple sources within a Playlist, and related use cases, introduced in Player iOS SDK version `v3` are not supported.
```

File: Example/Tests/Doubles/BitmovinPlayerTestDouble.swift
Comment (luckygoyal-bitmovin) [reply Example/Tests/Doubles/BitmovinPlayerTestDouble.swift]: Done

File: Example/Tests/Doubles/BitmovinPlayerTestDouble.swift
Comment (luckygoyal-bitmovin) [reply Example/Tests/Doubles/BitmovinPlayerTestDouble.swift]: Done

**Comment by luckygoyal-bitmovin** [comment]
> > > Yes, I saw that but still wanted to ask especially. I would propose to end the session when a `PlaylistTransitionEvent` occurs to not report state from subsequent Sources. And of course, stating this in the Limitations in the Readme.
> > 
> > 
> > Do you suggest this to ensure that if any customer uses Playlist API with this Conviva integration, then we should ensure that session atleast ends after first source in playlist? I would rather keep it unhandled yet.
> 
> Yes that would be my suggestion 🤔 Hmm we could also keep it unhandled, however, my thought is rather having the first source in a playlist tracked (and ended/stoped) properly than continuing tracking events from a subsequent source. But I leave it up to you at this point, you might have more insights into how this should be handled in the first version.
> 
> EDIT: especially the cross-source seeking might be a source or errors as a lot of source-related data changes however for the analytics integration it's 'just' a normal seek.

I have added logic to finish Conviva session on playlist transition as suggested.

**Review by stonko1994 — APPROVED** [review]
Nice job, well done 🙂 👍

File: Example/Podfile
Comment (dweinber) [inline Example/Podfile:6]: Let's use the stable version instead of the RC as it's available by now.

```suggestion
  pod 'BitmovinPlayer', '3.6.1'
```

File: README.md
Comment (dweinber) [inline README.md:24]: Nit: Should state Conviva integration version `>= 1.3.0 && < 2.0.0` IMO. It's implicit but might be worth making it explicit.

**Review by dweinber — APPROVED** [review]
Looks good! A few comments / small improvements, please check before merging (I approve nevertheless). We should at least update the version to not use the player RC.

File: BitmovinConvivaAnalytics/Classes/ContentMetadataBuilder.swift
Comment (luckygoyal-bitmovin) [reply BitmovinConvivaAnalytics/Classes/ContentMetadataBuilder.swift:91]: The order or setter and getter was changed to fix the build warnings.

File: README.md
Comment (luckygoyal-bitmovin) [reply README.md]: Done

File: BitmovinConvivaAnalytics/Classes/ContentMetadataBuilder.swift
Comment (dweinber) [reply BitmovinConvivaAnalytics/Classes/ContentMetadataBuilder.swift:91]: Interesting, didn't know that 👍

**Pull request merged**
